src/hooks/usePrevious.ts

```diff
@@ -0,0 +1,9 @@
+import { useEffect, useRef } from "react";
```

Comment:
Added as I thought I needed. Didn't need, but thought it might come in handy later.